tests/integration/targets/aci_l3out_logical_interface_profile_eigrp_profile/tasks/main.yml

```diff
@@ -0,0 +1,202 @@
+    register: nm_add_eigrpIfP
+
+  - name: Verify add_eigrpIfP
+    ansible.builtin.assert:
+      that:
+      - cm_add_eigrpIfP is changed
+      - nm_add_eigrpIfP is changed
+      - cm_add_eigrpIfP.previous == nm_add_eigrpIfP.previous == []
+      - cm_add_eigrpIfP.proposed.eigrpIfP.attributes.name == 'INTFS'
+      - cm_add_eigrpIfP.proposed.eigrpIfP.children[0].eigrpRsIfPol.attributes.tnEigrpIfPolName == 'default'
+      - nm_add_eigrpIfP.current[0].eigrpIfP.attributes.name == 'INTFS'
+      - nm_add_eigrpIfP.current[0].eigrpIfP.children[0].eigrpRsIfPol.attributes.tnEigrpIfPolName == 'default'
+      - nm_add_eigrpIfP.proposed.eigrpIfP.attributes.name == 'INTFS'
+      - nm_add_eigrpIfP.proposed.eigrpIfP.children[0].eigrpRsIfPol.attributes.tnEigrpIfPolName == 'default'
+
+  - name: Add eigrp profile again to check idempotency
+    cisco.aci.aci_l3out_logical_interface_profile_eigrp_profile: *eigrpIfP_present
+    register: nm_add_eigrpIfP_again
+
+  - name: Verify add_eigrpIfP_again
+    ansible.builtin.assert:
+      that:
+      - nm_add_eigrpIfP_again is not changed
+      - nm_add_eigrpIfP_again.previous == nm_add_eigrpIfP_again.current
+      - nm_add_eigrpIfP_again.proposed.eigrpIfP.attributes.name == 'INTFS'
+      - nm_add_eigrpIfP_again.proposed.eigrpIfP.children[0].eigrpRsIfPol.attributes.tnEigrpIfPolName == 'default'
+      - nm_add_eigrpIfP_again.current[0].eigrpIfP.attributes.name == 'INTFS'
+      - nm_add_eigrpIfP_again.current[0].eigrpIfP.children[0].eigrpRsIfPol.attributes.tnEigrpIfPolName == 'default'
+
+  # UPDATE INTERFACE PROFILE EIGRP PROFILE
+  - name: Update interface profile eigrp profile
+    cisco.aci.aci_l3out_logical_interface_profile_eigrp_profile: &eigrpIfP_update
+      <<: *eigrpIfP_present
+      eigrp_policy: ansible_eigrp_interface
+    register: update_eigrpIfP
+
+  - name: Update interface profile eigrp profile again
+    cisco.aci.aci_l3out_logical_interface_profile_eigrp_profile:
+      <<: *eigrpIfP_update
+    register: update_eigrpIfP_again
+
+  - name: Verify update_eigrpIfP
+    ansible.builtin.assert:
+      that:
+      - update_eigrpIfP is changed
+      - update_eigrpIfP.previous != []
+      - update_eigrpIfP.current[0].eigrpIfP.attributes.name == 'INTFS'
+      - update_eigrpIfP.current[0].eigrpIfP.children[0].eigrpRsIfPol.attributes.tnEigrpIfPolName == 'ansible_eigrp_interface'
+      - update_eigrpIfP.proposed.eigrpIfP.attributes.name == 'INTFS'
+      - update_eigrpIfP.proposed.eigrpIfP.children[0].eigrpRsIfPol.attributes.tnEigrpIfPolName == 'ansible_eigrp_interface'
+      - update_eigrpIfP_again is not changed
+      - update_eigrpIfP_again.previous == update_eigrpIfP_again.current
+
+  # QUERY INTERFACE PROFILE EIGRP PROFILE
+  - name: Query interface profile eigrp profile
+    cisco.aci.aci_l3out_logical_interface_profile_eigrp_profile:
+      <<: *eigrpIfP_update
+      state: query
+    register: query_eigrpIfP_profile
+
```

Comment:
could you check if you can create another l3out with a profile and do a query all? I suspect this would do a class query if no config besides * aci_info is provided and thus provide more results.

Reply:
Added, but this only works if I include the target_filter on the eigrpIfP class, mentioned above.